H/HDF5/generate-H5Tinit/generate-h5tinit.dockerfile

```diff
@@ -0,0 +1,27 @@
+# Run `docker build --file generate-h5tinit.dockerfile --build-arg cpuarch=amd64 --progress plain .`
+
+ARG cpuarch=amd64 # amd64, arm32v5, arm32v7, arm64v8, i386, mips64le, ppc64le, riscv64
+
+FROM ${cpuarch}/debian:11.6
+# FROM ${cpuarch}/ubuntu:18.04
+
+# Install packages
+ENV DEBIAN_FRONTEND=noninteractive
+RUN apt-get update && \
+    apt-get --yes --no-install-recommends install \
+        build-essential \
+        ca-certificates \
+        g++ \
+        gfortran \
+        wget
+
+# Download and build HDF5
+RUN wget https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.14/hdf5-1.14.0/src/hdf5-1.14.0.tar.gz
+RUN tar xzf hdf5-1.14.0.tar.gz
+WORKDIR hdf5-1.14.0
+RUN mkdir build
+WORKDIR build
+RUN ../configure --enable-c++ --enable-fortran
```

Comment:
@giordano what are the config options so that we can locate libaec?

Reply:
`libaec` provides a file `libsz.so`. I'm using `--with-szlib=${prefix}`.

Reply:
Ooh, you're describing the Dockerfile. We only need this for the floating point options; everything else works fine with cross-compiling.

Reply:
To clarify: In the Docker container we cannot located external dependencies. This container only provides a basic Debian environment, without any dependencies installed. It can only be used to query basic hardware and ABI features, such as floating point layouts. Everything else is done in the build script in the `build_tarballs.jl` file. The Dockerfile as is works fine – it needs to configure HDF5 for the particular CPU architecture and OS, and then extract a few autogenerated files plus the `config.status` that also contains some relevant information.